subprojects/gradle/room-config/src/main/kotlin/com/avito/android/plugin/incremental/RoomIncrementalKaptSupportCheckMode.kt

```diff
@@ -0,0 +1,36 @@
+package com.avito.android.plugin.incremental
+
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+
+internal enum class RoomIncrementalKaptSupportCheckMode {
+    NONE {
+        override fun check(project: Project) {
+            // do nothing
+        }
+    },
+    WARNING {
+        override fun check(project: Project) {
+            val supported = RoomIncrementalKaptSupportChecker(project).isSupported()
+            if (!supported) {
+                System.err.println(collectErrorMessage())
+            }
+        }
+    },
+    FAIL {
+        override fun check(project: Project) {
+            val supported = RoomIncrementalKaptSupportChecker(project).isSupported()
+            if (!supported) {
+                throw GradleException(collectErrorMessage())
+            }
+        }
+    };
+
+    abstract fun check(project: Project)
+
+    fun collectErrorMessage() = """
+        Incremental KAPT is turned on (kapt.incremental.apt=true) but Room does not support it in current conditions. 
+        You have to use JDK embedded in Android Studio 3.5.0-beta02 and higher.
```

Comment:
Can we point to any public issue or at least source code?

Reply:
Ok. I'll attach some information.

Reply:
Not actual. Move this check from this PR to another task.